lib/collection.js

```diff
@@ -1744,7 +1745,18 @@ Collection.prototype.aggregate = function(pipeline, options, callback) {
     pipeline = args;
   }
 
-  return aggregate(this.s.db, this, pipeline, options, callback);
+  const cursor = new AggregationCursor(
+    this.s.topology,
+    new AggregateOperation(this, pipeline, options),
+    options
+  );
+
+  if (typeof callback === 'function') {
```

Comment:
unrelated: I think this is a bad pattern left over from our pre-2.6 support. Can we:
1. add a TODO to get rid of this in the next breaking version
2. add a ticket to add a deprecation warning for this
3. add a ticket to remove this in 4.0?

Reply:
https://jira.mongodb.org/browse/NODE-2074